pan3d/dataset_builder.py

```diff
@@ -116,6 +141,37 @@ def data_range(self):
             return 0, 0
         return self.data_array.min(), self.data_array.max()
 
+    # -----------------------------------------------------
+    # UI bound methods
+    # -----------------------------------------------------
+
+    def coordinate_select_axis(self, coordinate_name, current_axis, new_axis, **kwargs):
+        if getattr(self.state, current_axis):
+            setattr(self.state, current_axis, None)
+        if new_axis and new_axis != "undefined":
+            setattr(self.state, new_axis, coordinate_name)
+        self.mesh_changed()
```

Comment:
Can you explain what you are trying to do here? That seems convoluted.

Reply:
https://trame.readthedocs.io/en/latest/core.state.html

Reply:
When a user selects an option in the "Assign Axis" dropdown for any coordinate, the previously assigned axis should be set to None (if there was a previous assignment) and the newly assigned axis should be set to the coordinate name. 

Is your suggestion to use `self.state.update(...)` instead of `setattr(self.state, ...)`?

Reply:
I would think something more like `self.state.new_axis = coordinate_name`

Reply:
unless I'm missing something

Reply:
Basically, I'm not sure why you use setattr and getattr...

Reply:
`new_axis` is a variable which will contain the name of a state attribute, it is not a state attribute itself. Its value would be one of `x_array`, `y_array`, `z_array`, or `t_array`.

Reply:
(the same applies to `current_axis`)

Reply:
then `self.state[new_axis] = coordinate_name`

Reply:
The first line of the State class API documentation

> Flexible dictionary managing a server shared state. Variables can be accessed with either the [] or . notation.

Reply:
And the usage of the state is also covered here https://kitware.github.io/trame/docs/cheatsheet.html#State

Reply:
Ok, I'll use bracket notation then, if that is preferred: changed in [5304ceb](https://github.com/Kitware/pan3d/pull/29/commits/5304ceb985f12378e6a2afde69136a19ba95a165)